File: frontend/src/services/ai.ts
import api from './api';

export interface LogAnalysisResponse {
  status: string;
  summary: string;
  root_cause: string;
  severity: string;
  suggested_fixes: string[];
  fix_commands: string[];
  confidence: string;
}

export interface AIRecommendation {
  id: string;
  category: string;
  title: string;
  description: string;
  impact: string;
  severity: 'low' | 'medium' | 'high';
  action: string;
}

export const analyzeLogsWithAI = async (logText: string, context?: string): Promise<LogAnalysisResponse> => {
  const response = await api.post<LogAnalysisResponse>('/ai/analyze-logs', {
    log_text: logText,
    context: context || 'general',
  });
  return response.data;
};

export const getAIRecommendations = async (): Promise<AIRecommendation[]> => {
  const response = await api.get<AIRecommendation[]>('/ai/recommendations');
  return response.data;
};

File: backend/app/api/v1/ai.py
from typing import Any, Dict, List, Optional
from fastapi import APIRouter, Depends
from pydantic import BaseModel
from app.api import deps
from app.models.user import User
from app.services.ai_service import ai_service

router = APIRouter()

class LogAnalysisRequest(BaseModel):
    log_text: str
    context: Optional[str] = "general"

class LogAnalysisResponse(BaseModel):
    status: str
    summary: str
    root_cause: str
    severity: str
    suggested_fixes: List[str]
    fix_commands: List[str]
    confidence: str

@router.post("/analyze-logs", response_model=LogAnalysisResponse)
def analyze_logs(
    request: LogAnalysisRequest,
    current_user: User = Depends(deps.get_current_active_user),
) -> Any:
    """
    AI-powered log analysis: parses raw logs and returns root cause diagnosis and fix commands.
    """
    return ai_service.analyze_logs(request.log_text, request.context or "general")

@router.get("/recommendations")
def get_ai_recommendations(
    current_user: User = Depends(deps.get_current_active_user),
) -> List[Dict[str, Any]]:
    """
    Get AI-driven cost, security, and performance recommendations for infrastructure.
    """
    return ai_service.get_recommendations()
